lib/ansible/plugins/lookup/file.py

```diff
@@ -70,7 +70,11 @@ def run(self, terms, variables=None, **kwargs):
             try:
                 if lookupfile:
                     b_contents, show_data = self._loader._get_file_contents(lookupfile)
-                    contents = to_text(b_contents, errors='surrogate_or_strict')
+                    try:
+                        contents = to_text(b_contents, errors='surrogate_or_strict')
+                    except UnicodeDecodeError:
+                        # Attempts utf-16 decode for files coming from Windows.
+                        contents = to_text(b_contents, encoding='utf-16', errors='surrogate_or_strict')
```

Comment:
This is fragile, and just a wild guess as to what we should do here.  Instead of this, we should be extending the functionality of the `file` lookup, to accept an `encoding` parameter.

Reply:
Had thought about making encoding a parameter but figured utf-8 and utf-16 covered both prevalent encodings. I implemented the option as you mentioned. Am just getting started with Ansible Contributions so I will have to look up how to add a unit test for this.

Reply:
I can't find how to add a unit test for the changes I made. Can you please provide documentation of how to submit unit tests for PRs made to the Ansible project? I updated the example block to include a debug msg which tests the option I have added for encoding.